### PR TITLE
[RLlib] Fix tf+GPU+CNN (Atari) (broken due to wrong tf session args).

### PR DESCRIPTION
**Pull request by sven1977**
Signed-off-by: sven1977 <svenmika1977@gmail.com>

Fix tf+GPU (broken due to wrong tf session args).

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ArturNiederfahrenhorst — APPROVED** [review]

**Comment by avnishn** [comment]
release tests running here:
https://buildkite.com/ray-project/release-tests-pr/builds/22824

File: rllib/utils/policy.py [hunk 1]
```diff
@@ -111,14 +111,17 @@ def create_policy_for_framework(
...
+                # Session creator function provided manually -> Use this one to
+                # create the tf1 session.
                 if session_creator:
                     sess = session_creator()
+                # Use a default session creator, based only on our `tf_session_args` in
+                # the config.
                 else:
                     sess = tf1.Session(
-                        config=tf1.ConfigProto(
-                            gpu_options=tf1.GPUOptions(allow_growth=True)
```
Comment (gjoliver) [inline rllib/utils/policy.py:119]: don't we need this for fractional gpu?
maybe it doesn't matter since this is tf1 ...

**Review by gjoliver — APPROVED** [review]

File: rllib/utils/policy.py (hunk 1 above)
Comment (sven1977) [reply rllib/utils/policy.py:119]: We still have this! It's "hidden" in our default values for `config.tf_session_args` (see algorithm_config.py).
And this is where this setting actually should be, not hidden somewhere here in this seemingly random python file :)

**Comment by kouroshHakha** [comment]
Didn't you also had a test that would check whether the LRU cache would work on GPU, why did that not catch this obvious failure?

**Comment by sven1977** [comment]
> Didn't you also had a test that would check whether the LRU cache would work on GPU, why did that not catch this obvious failure?

Good question. The problem only occurs when using a CNN :) So the condition for the error was:

- framework="tf" ("tf2" works fine)
- GPU
- RLlib default CNN model (e.g. for Atari env)

**Comment by sven1977** [comment]
What we should do next is to add all algorithm compilation tests to also run on the GPU in the CI. This would have caught it and gives us more protection against those GPU-related errors in the future. I'll prep a PR for this ...

**Pull request merged**